src/view-shot.cpp

```diff
@@ -29,11 +29,32 @@
 #include <wayfire/opengl.hpp>
 #include <wayfire/img.hpp>
 
+#include <ctime>
+
+static std::string replaceAll(std::string s, const std::string& from,
+    const std::string& to)
+{
+    for (unsigned i = 0; i < s.size();)
+    {
+        auto pos = s.find(from, i);
+        if (pos == std::string::npos)
+        {
+            return s;
+        }
+
+        s.replace(pos, from.size(), to);
+        i = pos + to.size();
+    }
+
+    return s;
+}
+
 class wayfire_view_shot : public wf::plugin_interface_t
 {
     const std::string transformer_name = "view_shot";
     wf::option_wrapper_t<wf::activatorbinding_t> capture_binding{"view-shot/capture"};
-    wf::option_wrapper_t<std::string> file_name{"view-shot/filename"};
+    wf::option_wrapper_t<std::string> file_format{"view-shot/filename"};
```

Comment:
I would rather leave this as 'file_name' because the option is filename and the default is `/tmp/snapshot.png`. However, I think I see that since this is being passed as the format argument of strftime(), it gives the user the option to insert format codes into the filename which would be replaced with the corresponding time component as per the man page. If this is correct, it should be documented in the xml <_long> for this option and the default file name changed to include the time in basic format.